File: E-mart_.Net/Backend/Services/CategoryService.cs
using E_mart.Models;
using E_mart.Repositories;
using Microsoft.EntityFrameworkCore;

namespace E_mart.Services
{
    public class CategoryService : ICategoryService
    {
        private readonly eMartDbContext _context;

        public CategoryService(eMartDbContext context)
        {
            _context = context;
        }

        public async Task<IEnumerable<Category>> GetAllCategories()
        {
            return await _context.Categories.ToListAsync();
        }

        public async Task<Category?> GetCategoryById(int id)
        {
            return await _context.Categories.FindAsync(id);
        }


    }
}

File: E-mart_java/Backend/src/main/java/com/example/Services/CategoryService.java
package com.example.Services;

import com.example.Models.Category;
import java.util.List;
import java.util.Optional;

public interface CategoryService {
    List<Category> getAllCategories();
    
    Optional<Category> getCategoryById(int id);
    
//    Category createCategory(Category category);
//    
//    Category updateCategory(int id, Category category);
//    
//    void deleteCategory(int id);
}
